OrchardCore.Commerce/Views/ProductPart.cshtml

```diff
@@ -1,18 +1,26 @@
 @model ProductPartViewModel
-@using OrchardCore.Commerce.Abstractions
-@{ 
-    int i = 0;
+
+@{
     var id = $"product-{Model.Sku}";
+    var attributes = Model
+        .Attributes
+        .Select((attribute, index) => (Attribute: attribute, Index: index));
```

Comment:
Interesting! Is this better than simply calling `.ToList()` and then using a `for` loop because we don't create another collection instance?

Reply:
Yes. `.Select` is a yielding method so it's only called in the `foreach` when a specific item is enumerated.

Reply:
I thought so. Nice stuff in .NET Core there is!